src/Compilers/CSharp/Test/Syntax/Parsing/DeconstructionTest.cs

```diff
@@ -600,18 +601,898 @@ void Foo()
+                                        {
+                                            N(SyntaxKind.VariableDeconstructionDeclarator);
+                                            {
+                                                N(SyntaxKind.OpenParenToken);
+                                                N(SyntaxKind.VariableDeclaration);
+                                                {
+                                                    N(SyntaxKind.VariableDeclarator);
+                                                    {
+                                                        N(SyntaxKind.IdentifierToken, "a");
+                                                    }
+                                                }
+                                                N(SyntaxKind.CommaToken);
+                                                N(SyntaxKind.VariableDeclaration);
+                                                {
+                                                    N(SyntaxKind.VariableDeclarator);
+                                                    {
+                                                        N(SyntaxKind.IdentifierToken, "b");
+                                                    }
+                                                }
+                                                N(SyntaxKind.CloseParenToken);
+                                            }
+                                        }
+                                        N(SyntaxKind.CommaToken);
+                                        N(SyntaxKind.VariableDeclaration);
+                                        {
+                                            N(SyntaxKind.VariableDeclarator);
+                                            {
+                                                N(SyntaxKind.IdentifierToken, "c");
+                                            }
+                                        }
+                                        N(SyntaxKind.CloseParenToken);
+                                        N(SyntaxKind.EqualsToken);
+                                        N(SyntaxKind.IdentifierName);
+                                        {
+                                            N(SyntaxKind.IdentifierToken, "foo");
+                                        }
+                                    }
+                                }
+                                N(SyntaxKind.SemicolonToken);
+                            }
+                            N(SyntaxKind.CloseBraceToken);
+                        }
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void VarMethodCall()
+        {
+            // `var(...);` is a method call
+            var tree = UsingTree(@"
+class C
+{
+    void Foo()
+    {
+        var(a, b);
```

Comment:
We're lucky that ref-returns have not shipped yet ;-)
LDM said it's ok for deconstruction to take over this ambiguous syntax: `var (x, y) = M();`

@VSadov Maybe we should have a specific error if you try to declare a ref-returning method called "var" to catch this sooner?